### ISSUE TITLE
compute local gradient

### ISSUE THREAD

**Issue by zhen0z** [sintefmath/JutulDarcy.jl#112]
Hi Olav,

Is it possible to compute the local gradient cell by cell in JutulDarcy? Here's what I mean, I get saturation from the simulation, the code can easily compute the gradient of saturation difference (observed saturation -predicted saturation) with respect to permeability. 

However, now I transform the saturation into something else, for example, seismic, then I get the new objective function loss = (seis_pred - seis_pred)^2.

I can compute the gradient of loss = (seis_pred - seis_pred)^2 with respect to saturation, now I think I need to get local gradient of dsw/dperm cell by cell to achieve the chain rule: 
dloss / dsat  * dsat/dperm.

Therefore, my question is: is it possible to compute the gradient of saturation to permeability cell by cell in JutulDarcy? Thanks.

Zhen

**Comment by moyner** [comment]
Hi!

There is no easy way to get the sens of saturation (in every cell) with respect to all permeabilities. You can get it one by one with adjoints (i.e. one saturation differentiated with respect to perm) but the general Jacobian you want is very dense and hard to obtain. The adjoint method we use is great for scalars (or a few scalars) with respect to all parameters but does not scale to many-to-many objectives functions.

**Comment by zhen0z** [comment]
I see, thanks for the explanations.
